q2studio/api/workspace.py

```diff
@@ -78,9 +78,10 @@ def get_artifacts():
 @fail_gracefully
 def create_artifact():
     request_body = request.get_json()
+    source_format = request_body.get('source_format')
```

Comment:
Instead of updating the workspace method, can you update `q2studio/api/formats.py`'s `get_importable_formats()` route in a manner similar to what you did above, with `get_importable_types()`?

Reply:
Is 9294032 what you had in mind?  `get_importable_formats` was/still is returning the correct values so I don't think that addresses the problem.

I suspect the problem is somewhere in JavaScript land.  Please see my failed attempt of implementing a solution discussed at https://stackoverflow.com/questions/43495696/how-to-set-a-default-value-in-react-select

It obviously doesn't work, but I think we need _something_ like this.  Unfortunately, my blind fumblings with JavaScript haven't amounted to anything.

```diff
modified   app/js/components/ReactSelect.jsx
@@ -14,16 +14,29 @@ import style from '!style-loader!css-loader!react-select-plus/dist/react-select-
 class ReactReduxSelect extends React.Component {
     constructor(props) {
         super(props);
-        this.state = {};
+        this.state = {selectedValue: '',};
+        this.handleChange = this.handleChange.bind(this);
+    };
+
+    componentDidMount() {
+        this.setState({
+            selectedValue: this.props.defaultValue,
+        })
+    }
+
+    handleChange(selectedOption) {
+        console.log(typeof selectedOption)
+        this.setState({selectedValue: selectedOption});
     }
+
     render() {
         return (
             <Select
                 name={this.props.name}
-                value={this.state.selected || this.props.placeholder}
+                value={this.props.options.filter(({value}) => value === this.state.selectedValue)}
                 options={this.props.options}
                 placeholder={this.props.placeholder}
-                onChange={(selectedValue) => { this.setState({ selected: selectedValue }); }}
+                onChange={this.handleChange}
             />
         );
     }
```

Reply:
Oh I thought you were updating things here in this method for stylistic changes - I didn't realize you were running into an issue here. Can you describe the issue and how to recreate it? As far as I can tell, this PR is working just fine, but I might not be catching the cases you're looking at.

Reply:
The issue is that `source_format` does not end up in the request body if no selection is made.  There is no default value.

The following implies the default value should be an empty string:
https://github.com/qiime2/q2studio/blob/015f858768aeadb7cb0b269ce46fe3a73649e045/app/js/containers/ArtifactGenerator.js#L29-L30

But that if statement never actually evaluates to true, which means that `source_format` never makes it into the request body.

You can reproduce simply by trying to import anything in q2studio.  `create_artifact()` expects the key `source_format` to be in the request body, but it's never in the request body.

Reply:
Perhaps something changed with one of our dependencies when you installed your development environment - a simple solution would be to update the version pins to pin to older versions (of what, i'm not sure, but my money is on redux and react). I'll take a look at this.